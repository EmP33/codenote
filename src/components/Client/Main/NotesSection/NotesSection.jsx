import { useCallback, useRef } from "react";
import { useParams } from "react-router-dom";
import styled from "styled-components";
import HomeNotes from "../HomeSection/HomeNotes/Notes";

import { Box, Button } from "@mui/material";

import { Ring } from "@uiball/loaders";

import { useDispatch, useSelector } from "react-redux";
import { addNote } from "../../../../store/user-slice";
import { v1 as uuidv1 } from "uuid";
import ReactEditor from "./ReactEditor/ReactEditor";

const NotesWrapper = styled.div`
  display: grid;
  overflow: hidden;
  grid-template-columns: repeat(2, 1fr);
  grid-template-rows: repeat(2, max-content);
  grid-row-gap: 3rem;
  grid-column-gap: 1rem;

  @media screen and (max-width: 750px) {
    grid-template-columns: 1fr;
  }
`;
const EditorContainer = styled.div`
  grid-column: 1/-1;
  display: grid;
  overflow: hidden;
  position: relative;
  grid-template-columns: 1fr;
  grid-template-rows: repeat(2, max-content);
  grid-row-gap: 3rem;
  grid-column-gap: 1rem;
`;

const NotesSection = () => {
  const params = useParams();
  const dispatch = useDispatch();
  const editorCore = useRef(null);
  const userData = useSelector((state) => state.user.userData);
  const user = useSelector((state) => state.user.user);
  const { uid } = user;
  const currentNote = userData.notes.find((note) => note.id === params.note);

  /* Saving the data from the editor. */
  const handleSave = useCallback(async () => {
    const savedData = await editorCore.current.save();

    dispatch(
      addNote(uid, {
        id: params.note === "new" ? uuidv1() : params.note,
        blocks: savedData.blocks,
        date: savedData.time,
        views: currentNote ? currentNote.views : 0,
      })
    );
  }, [dispatch, params.note, uid, currentNote]);

  // if (!userData.notes.length) return <h1>Loading..</h1>;

  return (
    <NotesWrapper>
      <HomeNotes />
      <EditorContainer>
        <Button
          variant="contained"
          color="secondary"
          size="large"
          sx={{
            position: "absolute",
            right: 0,
            top: 0,
            borderRadius: "20px 20px 0 0 ",
            width: "100%",
            margin: "0 auto",
            background: "var(--color-tertiary-dark)",
            cursor: "pointer",
          }}
          onClick={handleSave}
        >
          Save
        </Button>
        <Box
          sx={{
            background: "#fff",
            color: "#333",
            fontSize: 20,
            borderRadius: 5,
            padding: 1,
            marginTop: 4,
          }}
        >
          {userData.notes ? (
            <ReactEditor editorCore={editorCore} />
          ) : (
            <Box sx={{ width: "100%", textAlign: "center", p: 20 }}>
              <Ring size={40} lineWeight={5} speed={2} color="black" />
            </Box>
          )}
        </Box>
      </EditorContainer>
    </NotesWrapper>
  );
};

export default NotesSection;
