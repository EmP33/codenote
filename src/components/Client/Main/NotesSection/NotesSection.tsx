import React from "react";

const NotesSection = () => {
  return <div>NotesSection</div>;
};

export default NotesSection;
