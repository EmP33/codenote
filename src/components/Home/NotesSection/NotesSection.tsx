import React, { useEffect, useState } from "react";
import { Container, Typography, Grid } from "@mui/material";
import { motion } from "framer-motion";
import styled from "styled-components";
import ContentPoint from "../ContentPoint/ContentPoint";
import NoteElement from "../../elements/NoteElement/NoteElement";

const NotesWrapper = styled.section`
  width: 100%;
  min-height: 100vh;
  padding: 5rem 0 0 0;
  background: var(--color-tertiary);
  position: relative;

  &:after {
    content: "";
    width: 100%;
    height: 5rem;
    background: linear-gradient(
      0deg,
      rgba(78, 76, 165, 1) 50%,
      rgba(62, 61, 132, 1) 100%
    );
    position: absolute;
    top: 0;
    left: 0;
    z-index: 1;
  }
`;

const NotesSection: React.FC = () => {
  const [notesActions, setNotesActions] = useState(false);
  const [notesHeaderActions, setNotesHeaderActions] = useState(false);

  useEffect(() => {
    const handleScroll = () => {
      if (window.scrollY > 300) {
        setNotesHeaderActions(true);
      }
      if (window.scrollY > 400) {
        setNotesActions(true);
      }
    };
    window.addEventListener("scroll", handleScroll);
  }, []);

  return (
    <NotesWrapper>
      <Container
        maxWidth="xl"
        sx={{ display: "flex", flexDirection: "column" }}
      >
        <motion.span
          animate={{ opacity: notesHeaderActions ? 1 : 0 }}
          initial={{ opacity: 0 }}
        >
          <ContentPoint
            index="01"
            heading="Improve
        efficiency"
          />
          <Typography
            variant="h5"
            sx={{
              textAlign: "center",
              fontFamily: "var(--font-heading)",
              marginTop: { xs: 10, lg: 0 },
            }}
          >
            Improve your efficiency with our wonderful notes!
          </Typography>
        </motion.span>

        <Grid
          container
          rowSpacing={0}
          columnSpacing={{ xs: 0, sm: 2, md: 5 }}
          component={motion.div}
          sx={{
            justifyContent: "center",
            marginTop: 5,
            maxWidth: 1200,
            alignSelf: "center",
          }}
          animate={{
            opacity: notesActions ? 1 : 0,
            y: notesActions ? 0 : "100vh",
            transition: {
              type: "tween",
            },
          }}
          initial={{ opacity: 0, y: "100vh" }}
        >
          <Grid item>
            <NoteElement
              hoverOnNotes={true}
              title="Flight Details"
              date="5/3/22"
              content="Get to the airport by 7am before takeoff"
              id="n1"
            />
          </Grid>
          <Grid item>
            <NoteElement
              hoverOnNotes={true}
              title="Business Strategy"
              date="11/2/22"
              content="Goal Corner market for green homes in Emerald Heights area by specializing in modern"
              id="n2"
            />
          </Grid>
          <Grid item>
            <NoteElement
              hoverOnNotes={true}
              title="Today's Ideas"
              date="5/11/21"
              content="This is what I need to do today. First of all I have to go to barber to make my haircut, second go to work"
              id="n3"
            />
          </Grid>
          <Grid item>
            <NoteElement
              hoverOnNotes={true}
              title="Client Preferences"
              date="22/5/22"
              content="Regrouping to let James know which listings seem like the best it and..."
              id="n4"
            />
          </Grid>
          <Grid item>
            <NoteElement
              hoverOnNotes={true}
              title="Party Preparations"
              date="24/1/21"
              content="What I need to do to make party. First buy some groceries, second call a dj"
              id="n5"
            />
          </Grid>
        </Grid>
      </Container>
    </NotesWrapper>
  );
};

export default NotesSection;
