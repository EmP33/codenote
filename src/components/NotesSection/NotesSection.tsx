import React from "react";
import { Container, Typography, Grid } from "@mui/material";
import styled from "styled-components";
import ContentPoint from "../ContentPoint/ContentPoint";
import NoteElement from "./NoteElement/NoteElement";

const NotesWrapper = styled.section`
  width: 100%;
  min-height: 100vh;
  padding: 5rem 0 0 0;
  background: var(--color-secondary);
  position: relative;

  &:after {
    content: "";
    width: 100%;
    height: 5rem;
    background: linear-gradient(
      0deg,
      rgba(19, 15, 50, 1) 50%,
      rgba(62, 61, 132, 1) 100%
    );
    position: absolute;
    top: 0;
    left: 0;
    z-index: 1;
  }
`;

const NotesSection: React.FC = () => {
  return (
    <NotesWrapper>
      <Container
        maxWidth="xl"
        sx={{ display: "flex", flexDirection: "column" }}
      >
        <ContentPoint
          index="01"
          heading="Improve
        efficiency"
        />
        <Typography
          variant="h5"
          sx={{
            textAlign: "center",
            fontFamily: "var(--font-heading)",
            marginTop: { xs: 10, lg: 0 },
          }}
        >
          Improve your efficiency with our wonderful notes!
        </Typography>
        <Grid
          container
          rowSpacing={0}
          columnSpacing={5}
          sx={{
            justifyContent: "center",
            marginTop: 5,
            maxWidth: 1200,
            alignSelf: "center",
          }}
        >
          <Grid item>
            <NoteElement
              title="Flight Details"
              date="5/3/22"
              content="Get to the airport by 7am before takeoff"
            />
          </Grid>
          <Grid item>
            <NoteElement
              title="Business Strategy"
              date="11/2/22"
              content="Goal Corner market for green homes in Emerald Heights area by specializing in modern"
            />
          </Grid>
          <Grid item>
            <NoteElement
              title="Today's Ideas"
              date="5/11/21"
              content="This is what I need to do today. First of all I have to go to barber to make my haircut, second go to work"
            />
          </Grid>
          <Grid item>
            <NoteElement
              title="Client Preferences"
              date="22/5/22"
              content="Regrouping to let James know which listings seem like the best it and..."
            />
          </Grid>
          <Grid item>
            <NoteElement
              title="Party Preparations"
              date="24/1/21"
              content="What I need to do to make party. First buy some groceries, second call a dj"
            />
          </Grid>
          <Grid item>
            <NoteElement
              title="Party Preparations"
              date="24/1/21"
              content="What I need to do to make party. First buy some groceries, second call a dj"
            />
          </Grid>
        </Grid>
      </Container>
    </NotesWrapper>
  );
};

export default NotesSection;
