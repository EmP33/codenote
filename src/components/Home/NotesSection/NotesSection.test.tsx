import { shallow } from "enzyme";
import NotesSection from "./NotesSection";

it("should render NotesSection properly", () => {
  expect(shallow(<NotesSection />)).toMatchSnapshot();
});
