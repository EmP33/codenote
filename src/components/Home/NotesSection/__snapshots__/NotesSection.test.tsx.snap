// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render NotesSection properly 1`] = `
<styled.section>
  <ForwardRef(Container)
    maxWidth="xl"
    sx={
      Object {
        "display": "flex",
        "flexDirection": "column",
      }
    }
  >
    <ForwardRef(MotionComponent)
      animate={
        Object {
          "opacity": 0,
        }
      }
      initial={
        Object {
          "opacity": 0,
        }
      }
    >
      <ContentPoint
        heading="Improve
 efficiency"
        index="01"
      />
      <ForwardRef(Typography)
        sx={
          Object {
            "fontFamily": "var(--font-heading)",
            "marginTop": Object {
              "lg": 0,
              "xs": 10,
            },
            "textAlign": "center",
          }
        }
        variant="h5"
      >
        Improve your efficiency with our wonderful notes!
      </ForwardRef(Typography)>
    </ForwardRef(MotionComponent)>
    <ForwardRef(Grid)
      animate={
        Object {
          "opacity": 0,
          "transition": Object {
            "type": "tween",
          },
          "y": "100vh",
        }
      }
      columnSpacing={
        Object {
          "md": 5,
          "sm": 2,
          "xs": 0,
        }
      }
      component={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
        }
      }
      container={true}
      initial={
        Object {
          "opacity": 0,
          "y": "100vh",
        }
      }
      rowSpacing={0}
      sx={
        Object {
          "alignSelf": "center",
          "justifyContent": "center",
          "marginTop": 5,
          "maxWidth": 1200,
        }
      }
    >
      <ForwardRef(Grid)
        item={true}
      >
        <Memo(NoteElement)
          content="Get to the airport by 7am before takeoff"
          date="5/3/22"
          id="n1"
          title="Flight Details"
        />
      </ForwardRef(Grid)>
      <ForwardRef(Grid)
        item={true}
      >
        <Memo(NoteElement)
          content="Goal Corner market for green homes in Emerald Heights area by specializing in modern"
          date="11/2/22"
          id="n2"
          title="Business Strategy"
        />
      </ForwardRef(Grid)>
      <ForwardRef(Grid)
        item={true}
      >
        <Memo(NoteElement)
          content="This is what I need to do today. First of all I have to go to barber to make my haircut, second go to work"
          date="5/11/21"
          id="n3"
          title="Today's Ideas"
        />
      </ForwardRef(Grid)>
      <ForwardRef(Grid)
        item={true}
      >
        <Memo(NoteElement)
          content="Regrouping to let James know which listings seem like the best it and..."
          date="22/5/22"
          id="n4"
          title="Client Preferences"
        />
      </ForwardRef(Grid)>
      <ForwardRef(Grid)
        item={true}
      >
        <Memo(NoteElement)
          content="What I need to do to make party. First buy some groceries, second call a dj"
          date="24/1/21"
          id="n5"
          title="Party Preparations"
        />
      </ForwardRef(Grid)>
    </ForwardRef(Grid)>
  </ForwardRef(Container)>
</styled.section>
`;
